src/test/java/io/jenkins/plugins/analysis/warnings/recorder/BuildOnWorkerITest.java

```diff
@@ -0,0 +1,127 @@
+import org.jenkinsci.test.acceptance.docker.fixtures.SshdContainer;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Slave;
+import hudson.plugins.sshslaves.SSHLauncher;
+import hudson.slaves.DumbSlave;
+import hudson.slaves.EnvironmentVariablesNodeProperty;
+import hudson.slaves.EnvironmentVariablesNodeProperty.Entry;
+import hudson.tasks.Maven;
+import hudson.tasks.Shell;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.warnings.Gcc3;
+import io.jenkins.plugins.analysis.warnings.MavenConsole;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Tests a build on a worker.
+ *
+ * @author Michael Schmid, Raphael Furch
+ */
+abstract class BuildOnWorkerITest extends IntegrationTestWithJenkinsPerSuite {
+
+    void buildMavenProjectOnWorker(final Slave worker) {
+        FreeStyleProject project = createFreeStyleProject();
+        try {
+            project.setAssignedNode(worker);
+        }
+        catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+        buildSuccessfully(project);
+
+        project.getBuildersList().add(new Maven("verify", null));
+        copySingleFileToAgentWorkspace(worker, project, "affected-files/Main.java",
+                "src/main/java/hm/edu/hafner/analysis/Main.java");
+        copySingleFileToAgentWorkspace(worker, project, "pom.xml", "pom.xml");
+        enableWarnings(project, createTool(new MavenConsole(), ""));
+
+        scheduleSuccessfulBuild(project);
+
+        List<AnalysisResult> result = getAnalysisResults(Objects.requireNonNull(project.getLastBuild()));
+        assertThat(result.size()).isEqualTo(1);
+        assertThat(Objects.requireNonNull(project.getLastBuild().getBuiltOn()).getLabelString()).isEqualTo(
+                worker.getLabelString());
+    }
+
+    void buildMakeProjectOnWorker(final Slave worker) {
+        FreeStyleProject project = createFreeStyleProject();
+        try {
+            project.setAssignedNode(worker);
+        }
+        catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+        buildSuccessfully(project);
+
+        project.getBuildersList().add(new Shell("make"));
+        copySingleFileToAgentWorkspace(worker, project, "make-gcc/Makefile", "Makefile");
+        copySingleFileToAgentWorkspace(worker, project, "make-gcc/main.c", "main.c");
+        enableWarnings(project, createTool(new Gcc3(), ""));
+
+        buildSuccessfully(project);
+
+        List<AnalysisResult> results = getAnalysisResults(Objects.requireNonNull(project.getLastBuild()));
+        assertThat(results.size()).isEqualTo(1);
+        assertThat(Objects.requireNonNull(project.getLastBuild().getBuiltOn()).getLabelString()).isEqualTo(
+                worker.getLabelString());
+    }
+
+    DumbSlave setupDumpSlave() {
```

Comment:
Is this method not in the parent class yet?

Reply:
This method encapsulates the exception handling